client/src/worldgen.rs

```diff
@@ -256,6 +257,64 @@ impl ChunkParams {
         }
     }
 
+    fn generate_road_support(
+        &self,
+        center: na::Vector3<f64>,
+        coords: na::Vector3<u8>,
+    ) -> Option<Material> {
+        let plane = -Plane::from(Side::B);
+        let horizontal_distance = plane.elevation(center, self.chunk);
+
+        if horizontal_distance < 0.3 {
+            let mat = self
+                .generate_trussing(coords)
+                .unwrap_or_else(|| Material::WoodPlanks);
+            if mat == Material::Void {
+                None
+            } else {
+                Some(mat)
+            }
+        } else {
+            None
+        }
+    }
+
+    ///carve out a truss-shaped hole
+    fn generate_trussing(&self, coords: na::Vector3<u8>) -> Option<Material> {
+        //Generates planar diagonals, but corner is offset
+        let mut criteria_met = 0_u32;
+        let x = coords[0];
+        let y = coords[1];
+        let z = coords[2];
+
+        //straight lines
+        if x == 8 {
```

Comment:
Let's not hardcode coordinates; that'll be very fragile as we adjust the chunk size in the future. Instead, compute the desired value from `dimension` dynamically; `2 * dimension / 3` perhaps? Opened #97 to help with this.